Dockerfile

```diff
@@ -30,4 +30,9 @@ COPY --from=build /home/builder/autotest-runner.jar ./
 # Copy cached dependencies
 COPY --from=build /home/gradle /home/gradle
 
+RUN pkill -f '.*GradleDaemon.*'
+RUN rm -rf ~/.gradle/daemon
+
+ENV GRADLE_OPTS="--add-opens java.base/java.util=ALL-UNNAMED --add-opens java.base/java.lang=ALL-UNNAMED --add-opens java.base/java.lang.invoke=ALL-UNNAMED --add-opens java.base/java.util=ALL-UNNAMED --add-opens java.prefs/java.util.prefs=ALL-UNNAMED --add-opens java.prefs/java.util.prefs=ALL-UNNAMED --add-opens java.base/java.nio.charset=ALL-UNNAMED --add-opens java.base/java.net=ALL-UNNAMED --add-opens java.base/java.util.concurrent.atomic=ALL-UNNAMED -Xms256m -Xmx2048M -Dfile.encoding=UTF-8 -Duser.country=US -Duser.language=en -Duser.variant"
```

Comment:
This is one core change, basically, the `--no-daemon` option passed from the command line was ineffective causing the startup of a daemon every execution (it was dominating the used time).
I struggled with this for a long and the only "solution/workaround" I've found is this:
https://github.com/gradle/gradle/issues/2660#issuecomment-602628786

This is ugly and weak (especially maintenance-wise) but is the only thing that actually works.

Reply:
Ah yes, my investigations pointed out that no daemon was used, but I didn't know how to fix it. Great work!

Reply:
@andreaTP It looks like in the above changes you're killing all Gradle daemons, right? If so, would a running Gradle daemon that can be re-used when running the tests be faster than `--no-daemon`? In some tracks, what I did is to build a "dummy" project to pre-load/cache things. I have no idea if that makes sense here though.

Reply:
> what I did is to build a "dummy" project to pre-load/cache things

do you mind elaborating or providing a link? might be interesting to investigate further this path!

Reply:
Sure! For Purescript, we have a regular exercise https://github.com/exercism/purescript-test-runner/tree/main/pre-compiled that we copy into the Docker image and then build/compile https://github.com/exercism/purescript-test-runner/blob/main/Dockerfile#L14. Doing a build/compile forces the Docker container to at least have loaded all the infrastructure once. Whether that is beneficial depends on the runtime that is used.

I tried using this approach with the Java test runner too, as I hoped that being able to re-use an existing Daemon that had already done some work might allow things to be sped up later. I couldn't get the `gradle test` command to re-use the started Gradle daemon, but I think you could with the GRADLE_OPTS specified.

Reply:
> I couldn't get the gradle test command to re-use the started Gradle daemon, but I think you could with the GRADLE_OPTS specified.

correct but it doesn't seem to produce great results ...

Reply:
Well, that could be because there actually isn't any Gradle daemon already running, right?

Reply:
correct